### DISCUSSION TITLE
Styling a progress bar while using WidgetTheme

### DISCUSSION THREAD

**Discussion by CramBL** [fltk-rs/fltk-rs#1473] [Q&A] [answered]
Hi, 

It appears to me that `WidgetTheme` overrides colors on misc::Progress (progress bar), basically making the progress bar useless.

The following example shows a window with a white progress bar that fills with green, going from 0 ->100% in a few seconds.

Commenting in `// theme.apply();` will style the progress bar in the style of `ThemeType::Dark` and now it doesn't fill with green and is just a box with text that slowly changes once the progress bar goes to 100. It's no longer really a progress bar.

```rust
// Cargo.toml
// [dependencies]
// fltk = "1.4"
// fltk-theme = "0.7"

use fltk::{
    app::{self, *},
    enums::Color,
    misc::Progress,
    prelude::*,
    window::*,
};
use fltk_theme::{ThemeType, WidgetTheme};

fn main() {
    let app = App::default();
    let theme = WidgetTheme::new(ThemeType::Dark);
    // theme.apply(); // comment in this to style the app and make the progress bar useless

    let mut wind = Window::new(100, 100, 400, 300, "Progress Bar");

    let mut progress = Progress::new(50, 200, 300, 30, "Progress bar title");
    progress.set_color(Color::White);
    progress.set_selection_color(Color::Green);
    progress.set_maximum(100.);
    progress.set_value(0.);

    wind.end();
    wind.show();

    std::thread::spawn(move || {
        for i in 0..=100 {
            progress.set_value(i as f64);
            progress.set_label(&format!("{i}%"));
            app::awake();
            app::sleep(0.05);
        }
    });

    app.run().unwrap();
}
```

Am I using the `WidgetTheme` in a wrong way? Is there anyway to get a proper progress bar while using `WidgetTheme`?

Thanks in advance 😃

**Comment by MoAlyousef** [comment, marked as answer]
Hello

The themes work by basically overriding the drawing routines of commonly used frame types. Coincidentally, Progress uses a DownBox which is what's also commonly used for input/text widgets. You can try changing the Progress' frame type:
```rust
    progress.set_frame(FrameType::FlatBox);
    progress.set_color(Color::Background2);
    progress.set_selection_color(Color::Green);
```
We also change the main color to Background2 just so the text would appear within the Progess, since Progress has no set_text_color() method.

**Reply by CramBL** [reply]
Thank you so much! Perfect! 🥳
